packages/server/index.js

```diff
@@ -27,4 +25,10 @@ process.enablePromiseAPIs = process.env.CYPRESS_ENV !== 'production'
 
 require('./lib/util/suppress_unauthorized_warning').suppress()
 
-module.exports = require('./lib/cypress').start(process.argv)
+const nodeOptions = require('./lib/util/node_options')
+
+if (!nodeOptions.maybeFork()) {
+  nodeOptions.reset()
+
+  module.exports = require('./lib/cypress').start(process.argv)
+}
```

Comment:
I would separate out the methods that return whether the fork is needed from the fork itself - and reverse the `if` condition to only do that when the fork is necessary (since that's the rare, minority case). 

As it's written - the logic implies the opposite (that this is the majority). I would also separate out the mutation and make the check pure and idempotent.

```js
// if our parent process was launched without the correct
// NODE_OPTIONS environment variable then fork a new child
// process with the fixed options
if (nodeOptions.needsFork()) {
  return nodeOptions.forkWithCorrectOptions()
}

// otherwise preserve the original options
nodeOptions.preserveOriginalOptions()

module.exports = require('./lib/cypress').start(process.argv)
```

Reply:
It's reversed because we can't return here since it's not a function, so for Cypress not to start, it needs to be in some kind of control structure. I can make it an if/else, but I think those are frowned upon in this project, so I'll just make a `launch` function.